File: algorithms/Sum of Left Leaves/Sum of Left Leaves.cpp
class Solution {
    int sum=0;
public:
    bool solve(TreeNode* root){
        if(root==NULL) return false;
        if(root->left==NULL && root->right==NULL) return true;
        if(solve(root->left)) sum+=root->left->val;
        solve(root->right);
        return false;
    }
    int sumOfLeftLeaves(TreeNode* root) {
        solve(root);
        return sum;
    }
};

File: scripts/Sum of Left Leaves/Sum of Left Leaves.js
var sumOfLeftLeaves = function(root) {
    let total = 0;
    
    const go = (node, isLeft) => {
        if (isLeft && !node.left && !node.right) {
            total += node.val;
            return;
        }
        if (node.left) go(node.left, true);
        if (node.right) go(node.right, false)
    }
    
    go(root, false)
    
    return total;
};
